From 3591894908fcac35d66a9a1f8b8885457df90f58 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Thu, 8 Nov 2018 15:52:08 +0200
Subject: [PATCH 5076/5242] MLK-20189-3: ASoC: fsl: library_load: Remove dead
 code

commit  984f676b90308bd04c8d6a4986593756537ec465 from
https://source.codeaurora.org/external/imx/linux-imx.git

At this point err is always 0. So, just remove the check.

Fixes: CID1477380

Reviewed-by: S.j. Wang <shengjiu.wang@nxp.com>
Reviewed-by: Cosmin-Gabriel Samoila <cosmin.samoila@nxp.com>
Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/fsl_dsp_library_load.c |    5 -----
 1 file changed, 5 deletions(-)

diff --git a/sound/soc/fsl/fsl_dsp_library_load.c b/sound/soc/fsl/fsl_dsp_library_load.c
index 37bce3a..11bb939 100644
--- a/sound/soc/fsl/fsl_dsp_library_load.c
+++ b/sound/soc/fsl/fsl_dsp_library_load.c
@@ -438,11 +438,6 @@ static int get_dyn_info(Elf32_Ehdr *eheader,
 				  xtlib_globals->byteswap),
 		  lib_info);
 
-	if (err != XTLIB_NO_ERR) {
-		xtlib_globals->err = err;
-		return 0;
-	}
-
 	return (xt_ptr)xtlib_host_word((Elf32_Word)info->start_sym,
 				       xtlib_globals->byteswap);
 }
-- 
1.7.9.5

